### DISCUSSION TITLE
Useless code

### DISCUSSION THREAD

**Discussion by MallauryGerard** [bagisto/bagisto#7340] [General]
Hello,

I was editing a file and I came across this code which is useless...

```
@php
    $showCompare = core()->getConfigData('general.content.shop.compare_option') == "1" ? true : false
@endphp

@php
    $showWishlist = core()->getConfigData('general.content.shop.wishlist_option') == "1" ? true : false;
@endphp

@if ($showWishlist)
    <li>
        <a href="{{ route('customer.wishlist.index') }}">
            {{ __('shop::app.header.wishlist') }}
        </a>
    </li>
@endif

@if ($showCompare)
    <li>
        <a href="{{ route('velocity.product.compare') }}">
            {{ __('shop::app.customer.compare.text') }}
        </a>
    </li>
@endif
```

Regardless of the PHP version, it is not necessary to cast a string to a boolean.
And if you look for `== "1" ? true` in your IDE, you will find this code 30 times...

**Solution 1:**

```
@if (core()->getConfigData('general.content.shop.wishlist_option'))
    <li>
        <a href="{{ route('customer.wishlist.index') }}">
            {{ __('shop::app.header.wishlist') }}
        </a>
    </li>
@endif
```

**Solution 2:**
A cleaner way to do that would be to create your own blade directive. This way:
```
@isConfigTrue('general.content.shop.wishlist_option')
```

Thanks!

**Comment by devansh-webkul** [comment]
We need to improve their method for its return types. Currently, there are three possible types: "1", "0", and nullable itself. Both "1" and "0" are not boolean values; they are string values. That's why you commonly see this in most places. We are already aware of these issues, and we are trying to make the return type cleaner to eliminate this inconsistency.

**Reply by MallauryGerard** [reply]
In PHP, the string "0" is actually evaluated as false.
I agree, tt would be cleaner to standardise the type of values. 
But whatever happens, PHP will interpret true/false and "1"/"0" correctly.
